File: src/test/java/com/github/alexandrecarlton/idea/settings/applier/impl/editor/spelling/SpellingSettingsApplierTest.java
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.spelling;

import static org.assertj.core.api.Assertions.assertThat;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.fixtures.IdeaSettingsTestFixture;
import com.github.alexandrecarlton.idea.settings.layout.editor.spelling.ImmutableSpellingSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.spelling.SpellingSettings;
import com.intellij.spellchecker.settings.SpellCheckerSettings;
import java.nio.file.Path;
import java.nio.file.Paths;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class SpellingSettingsApplierTest extends IdeaSettingsTestFixture {

  private SettingsApplier<SpellingSettings> settingsApplier;
  private SpellCheckerSettings spellCheckerSettings;

  @Before
  public void setUp() {
    spellCheckerSettings = SpellCheckerSettings.getInstance(project);
    settingsApplier = new SpellingSettingsApplier(spellCheckerSettings);
  }

  @After
  public void tearDown() {
    spellCheckerSettings.getCustomDictionariesPaths().clear();
  }

  @Test
  public void dictionaryRelativeToProjectApplied() {
    Path dict = Paths.get(project.getBasePath()).resolve("dict.dic");
    settingsApplier.apply(ImmutableSpellingSettings.builder()
        .addDictionaries(dict)
        .build());
    assertThat(spellCheckerSettings.getCustomDictionariesPaths()).containsOnly(dict.toString());
  }

  @Test
  public void absoluteDictionaryApplied() {
    settingsApplier.apply(ImmutableSpellingSettings.builder()
        .addDictionaries(Paths.get("/tmp/dict.dic"))
        .build());
    assertThat(spellCheckerSettings.getCustomDictionariesPaths()).containsOnly("/tmp/dict.dic");
  }
}

File: src/test/kotlin/com/github/alexandrecarlton/idea/settings/applier/impl/editor/spelling/SpellingSettingsApplierTest.kt
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.spelling

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier
import com.github.alexandrecarlton.idea.settings.fixtures.IdeaSettingsTestFixture
import com.github.alexandrecarlton.idea.settings.layout.editor.spelling.SpellingSettings
import com.intellij.spellchecker.settings.SpellCheckerSettings
import org.assertj.core.api.Assertions.assertThat
import org.junit.After
import org.junit.Before
import org.junit.Test
import java.io.File

class SpellingSettingsApplierTest : IdeaSettingsTestFixture() {

    private lateinit var settingsApplier: SettingsApplier<SpellingSettings>
    private lateinit var spellCheckerSettings: SpellCheckerSettings

    @Before
    public override fun setUp() {
        spellCheckerSettings = SpellCheckerSettings.getInstance(project)
        settingsApplier = SpellingSettingsApplier(spellCheckerSettings)
    }

    @After
    public override fun tearDown() {
        spellCheckerSettings.customDictionariesPaths.clear()
    }

    @Test
    fun dictionaryRelativeToProjectApplied() {
        val dict = File(project.basePath).resolve("dict.dic")
        settingsApplier.apply(SpellingSettings(dictionaries = listOf(dict)))
        assertThat(spellCheckerSettings.customDictionariesPaths).containsOnly(dict.toString())
    }

    @Test
    fun absoluteDictionaryApplied() {
        settingsApplier.apply(SpellingSettings(dictionaries = listOf(File("/tmp/dict.dic"))))
        assertThat(spellCheckerSettings.customDictionariesPaths).containsOnly("/tmp/dict.dic")
    }
}
